doc/neps/nep-0049.rst

```diff
@@ -153,12 +156,17 @@ NumPy C-API functions
    hooks. All the function pointers must be filled in, ``NULL`` is not
    accepted.
 
-.. c:function:: const char * PyDataMem_GetHandlerName(PyArrayObject *obj)
+.. c:function:: const PyDataMem_Handler * PyDataMem_GetHandler(PyArrayObject *obj)
 
-   Return the const char name of the ``PyDataMem_Handler`` used by the
+   Return the ``PyDataMem_Handler`` used by the
    ``PyArrayObject``. If ``NULL``, return the name of the current global policy
    that will be used to allocate data for the next ``PyArrayObject``.
 
+``PyDataMem_Handler`` thread safety and lifetime
+================================================
+As an implementation detail, currently the ``Set`` function will store the
+``PyDataMem_Handler`` in a ``PyCapsule`` as a :py:class:`ContextVar
+<contextvars.ContextVar>`.
```

Comment:
The fact that the semantics match those of `ContextVar`s does not sound like an implementation detail; only the fact it contains a `PyCapsule` is. Perhaps something more like:
```suggestion
``PyDataMem_Handler`` thread safety and lifetime
================================================
The active handler is stored in the current :py:class;`~contextvars.Context`
via a :py:class:`~contextvars.ContextVar`. This ensures it is both thread- and
async-safe.

As an implementation detail, currently this ContextVar contains a ``PyCapsule``
object storing a pointer to a ``PyDataMem_Handler``, but this should not be relied
upon.
```